x-pack/plugins/apm/public/components/shared/Links/apm/service_inventory_link.tsx

```diff
@@ -10,16 +10,21 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import React from 'react';
-import { APMLink, APMLinkExtendProps } from './APMLink';
-import { useUrlParams } from '../../../../hooks/useUrlParams';
 import { pickKeys } from '../../../../../common/utils/pick_keys';
+import { useUrlParams } from '../../../../hooks/useUrlParams';
+import { APMQueryParams } from '../url_helpers';
+import { APMLink, APMLinkExtendProps, useAPMHref } from './APMLink';
+
+const persistedFilters: Array<keyof APMQueryParams> = ['host', 'agentName'];
 
-function ServiceInventoryLink(props: APMLinkExtendProps) {
+export function useServiceInventoryHref() {
```

Comment:
Should the filename be changed to `use_service_inventory_link.tsx`?

Reply:
I opened #83196 to look at this.